src/modules/universe/RadixNode.ts

```diff
@@ -2,21 +2,25 @@ import { RadixNodeInfo } from '../..';
 import Long from 'long';
 
 export default class RadixNode {
+    public static SHARD_CHUNK_HALF_RANGE = Long.fromValue(1).shl(43)
+
     constructor(readonly nodeInfo: RadixNodeInfo, readonly wsAddress: string, readonly httpAddress: string) {}
 
     public canServiceShard(shard: Long): boolean {
+        const remainder = shard.mod(RadixNode.SHARD_CHUNK_HALF_RANGE)
+
         if (this.nodeInfo.system) {
-            const low = Long.fromValue(this.nodeInfo.system.shards.low)
-            const high = Long.fromValue(this.nodeInfo.system.shards.high)
+            const low = Long.fromValue(this.nodeInfo.system.shards.range.low)
+            const high = Long.fromValue(this.nodeInfo.system.shards.range.high)
 
             if (high.lessThan(low)) {
                 // Wrap around
```

Comment:
I'm curious, I have not seen Java lib wrapping around the range, is this documented somewhere (WP, Confluence?), maybe I've just miseed that Java wraps...?

Reply:
No, Java doesn't wrap, I coded this before the current implementation. The core enforces `low<high`, not sure if we're going to get back to wraparound, I think we should though

Reply:
I also enforce low<high, we might wanna add a ticket about this so we consolidate the three libraries (possible forth: Kotlin) :)

Reply:
Would be good to get a document describing sharding in detail, how it works now and how it will work by launch

Reply:
agreed!